packages/adapter-components/src/elements/instance_elements.ts

```
@@ -82,11 +82,16 @@ export const toBasicInstance = async ({
     transformationDefaultConfig,
   )
 
-  const nameParts = idFields.map(field => _.get(entry, field))
-  if (nameParts.includes(undefined)) {
+  const nameParts = _.isEmpty(idFields)
```

I think we should just fail that as invalid input - an empty `idFields` spec is invalid and we shouldn't silently handle it.
We can add that validation for `transformation` - e.g. in https://github.com/salto-io/salto/blob/main/packages/adapter-components/src/config/transformation.ts#L152 (and in the validation for the defaults used by all types as well)